src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/DefaultModelMetadataProvider.cs

```diff
@@ -173,12 +159,17 @@ public TypeCache()
             }
         }
 
-        private class PropertiesCache : ConcurrentDictionary<ModelMetadataIdentity, DefaultMetadataDetailsCache[]>
+        private struct ModelMetadataCacheEntry
         {
-            public PropertiesCache()
-                : base(ModelMetadataIdentityComparer.Instance)
+            public ModelMetadataCacheEntry(ModelMetadata metadata, DefaultMetadataDetails details)
             {
+                Metadata = metadata;
+                Details = details;
             }
+
+            public ModelMetadata Metadata { get; private set; }
+
+            public DefaultMetadataDetails Details { get; private set; }
```

Comment:
I know someone is going to suggest that I change this code in a small way in order to earn more PranavPoints. Let me preempt you.... this is a struct.